app/jobs/cancel_reservations_job.rb

```diff
@@ -0,0 +1,23 @@
+class CancelReservationsJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    reservations.each do |reservation|
+      cancel_reservation(reservation) if not_confirmed_on_time?(reservation)
+    end
+  end
+
+  private
+
+  def not_confirmed_on_time?(reservation)
+    reservation.seance.start_time <= 30.minutes.from_now
+  end
+
+  def reservations
```

Comment:
```suggestion
  def pending_reservations
```

it even brings more info to readers :wink:

Reply:
True, changed that :D